client/src/modules/stock/entry/entry.js

```diff
@@ -658,24 +667,35 @@ function StockEntryController(
    * @description [grid] initialize each cell of defined rows with value
    */
   function buildStockLine(line) {
-    const inventory = inventoryStore.get(line.inventory_uuid);
-    const entryDate = vm.movement.date || Date();
-    line.code = inventory.code;
-    line.label = inventory.label;
-    line.unit_cost = inventory.price;
-    line.quantity = 0;
-    line.cost = line.quantity * line.unit_cost;
-    line.expiration_date = entryDate;
-    line.unit = inventory.unit;
-    line.tracking_expiration = inventory.tracking_expiration;
-    setInitialized(line);
 
     // Store the non-expired candidate lots for this inventory code
     Lots.candidates({ inventory_uuid : line.inventory_uuid, date : vm.movement.date })
       .then((lots) => {
         line.availableLots = lots;
         line.candidateLots = lots.filter(lot => !lot.expired);
-      });
+
+        const inventory = inventoryStore.get(line.inventory_uuid);
+        const entryDate = vm.movement.date || Date();
+        line.code = inventory.code;
+        line.label = inventory.label;
+        line.unit_cost = inventory.price;
+        line.quantity = 0;
+        line.cost = line.quantity * line.unit_cost;
+        line.expiration_date = entryDate;
+        line.unit = inventory.unit;
+        line.tracking_expiration = inventory.tracking_expiration;
+        setInitialized(line);
+
+        return vm.depot && vm.depot.uuid ? Stock.inventories.read(null, {
+          inventory_uuid : line.inventory_uuid,
+          depot_uuid : vm.depot.uuid,
+        }) : {};
+      })
+      .then(stockInventory => {
+        if (!stockInventory.length) { return; }
```

Comment:
When would this occur?  And should we warn the user when it does?

Reply:
The `Stock.inventories.read(null, { inventory_uuid : line.inventory_uuid, depot_uuid : vm.depot.uuid })` function returns an array; the case where `!stockInventory.length` can be true is when the inventory_uuid doesn't exist in the database or when there is nothing as stock in the database for the given depot_uuid; All these cases are corner cases because in normal usage we think these cases cannot happen.

I agree with the idea to warn the user when it happens